test/diExports.ts

```diff
@@ -1,13 +1,19 @@
+// import { TEST_BROWSER_DYNAMIC_APPLICATION_PROVIDERS } from '@angular/platform-browser-dynamic/testing';
+// import { TEST_BROWSER_DYNAMIC_PLATFORM_PROVIDERS } from '@angular/platform-browser-dynamic/testing';
 import { provide, Type }                              from '@angular/core';
 import { ComponentFixture, TestComponentBuilder }     from '@angular/compiler/testing';
-import { inject, async }                              from '@angular/core/testing';
+import { inject, async, setBaseTestProviders}                              from '@angular/core/testing';
 import { disableDeprecatedForms, provideForms, FormControl } from '@angular/forms';
 import { App, Config, Form, NavController, Platform } from 'ionic-angular';
 import { ConfigMock, NavMock }                        from './mocks';
 import { Utils }                                      from '../app/services/utils';
 export { TestUtils }                                  from './testUtils';
 
+// Your patch suggest I use setBaseTestProviders here?
```

Comment:
I didn't get that issue, but you can always use resetBaseTestProviders
before the set.

On 26 Aug 2016 00:39, "cesco" <notifications@github.com> wrote:

> In test/diExports.ts
> <https://github.com/lathonez/clicker/pull/131#discussion_r76255563>:
>
> >  import { disableDeprecatedForms, provideForms, FormControl } from '@angular/forms';
> >  import { App, Config, Form, NavController, Platform } from 'ionic-angular';
> >  import { ConfigMock, NavMock }                        from './mocks';
> >  import { Utils }                                      from '../app/services/utils';
> >  export { TestUtils }                                  from './testUtils';
> >
> > +// Your patch suggest I use setBaseTestProviders here?
>
> @lathonez <https://github.com/lathonez> your patch suggests I use
> setBaseTestProviders here instead of app.spec.ts.
> I get this error
>
> Failed: addProviders can't be called after the injector has been already cr
> eated for this test. This is most likely because you've already used the
> injector t
> o inject a beforeEach or the current it function.
>
> Error: addProviders can't be called after the injector has been already cre
> ated for this test. This is most likely because you've already used the
> injector to
> inject a beforeEach or the current it function.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lathonez/clicker/pull/131/files/6020f8148377bc7b8877350b074679f61b73b9ab#r76255563>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AG5tSJw592b9xOQe4HWonZj4GkGTkzvZks5qjajlgaJpZM4JjxWv>
> .
>